### PR TITLE
Cypher builder connections

### LINKED ISSUE #1782: Nested connection where SINGLE and SOME not working as expected
**Describe the bug**
Related to #1221, triple nested relationships fail to use connection filters `_SOME` and `_SINGLE`.

**query**
```graphql
                query (
                    $where: MainWhere = { current: true }
                    $connectionWhere: RelationPropsWhere = { current: true }
                ) {
                    mains(where: $where) {
                        id
                        mainConnection(where: { edge: $connectionWhere }) {
                            edges {
                                node {
                                    architectureConnection(where: { edge: $connectionWhere }) {
                                        edges {
                                            node {
                                                nameDetailsConnection(where: { edge: $connectionWhere }) {
                                                    edges {
                                                        node {
                                                            fullName
                                                        }
                                                    }
                                                }
                                            }
                                        }
                                    }
                                }
                            }
                        }
                    }
                }
```

**params**
```javascript
          where: {
                current: true,
                mainConnection_SINGLE: {
                    node: {
                        architectureConnection_SOME: {
                            node: {
                                nameDetailsConnection: {
                                    node: {
                                        fullName: "MHA",
                                    },
                                },
                            },
                        },
                    },
                },
            },
            connectionWhere: {
                current: true,
            },
```

### LINKED ISSUE #1783: Missing parameters on the connection where for implicit "and"
**Describe the bug**
Missing parameters on the connection where for implicit "and"

**Query**
```graphql
  query (
      $where: SeriesWhere
      $connectionWhere: RelationPropsWhere
  ) {
     series(where: $where) {
          id
          nameDetailsConnection(where: {edge: $connectionWhere}) {
              edges {
                  node {
                      fullName
                  }
              }
          }
          architectureConnection(where: { edge: $connectionWhere }) {
              edges {
                  node {
                      nameDetailsConnection(where: { edge: $connectionWhere }) {
                          edges {
                              node {
                                  fullName
                              }
                          }
                      }
                  }
              }
          }
      }
  }
```

**Params**
```js
  where: {
      current: true,

      architectureConnection_SINGLE: {
          edge: {
              current: true,
          },
          node: {
              nameDetailsConnection: {
                  edge: {
                      current: true,
                  },
                  node: {
                      fullName: "fullname1",
                  },
              },
          },
      },
  },
  connectionWhere: {
      current: true,
  },
```

The result cypher params are missing the fullName filter

### PR DESCRIPTION
**Pull request by angrykoala**
# Description

This PR moves the connections where translation to cypher builder and removes one instance of runFirstColumn.

This PR depends on #1769 

Fixes #1782 
Fixes #1783

Slight improvement on performance for deeply nested connection filters

### CODE REVIEW

File: packages/graphql/tests/integration/issues/1221.int.test.ts [hunk 1]
```diff
@@ -480,7 +480,9 @@ describe("https://github.com/neo4j/graphql/issues/1221", () => {
                 current: true,
                 mainConnection_SINGLE: {
                     node: {
-                        architectureConnection: {
+                        // TODO: is SINGLE correct?
+                        architectureConnection_SINGLE: {
```
Comment (angrykoala) [inline packages/graphql/tests/integration/issues/1221.int.test.ts:484]: NOTE: This may be a bug fixed

File: .changeset/large-hounds-stare.md [hunk 2]
```diff
@@ -0,0 +1,5 @@
+---
+"@neo4j/graphql": patch
+---
+
+Fix nested connection filters using SINGLE and SOME
```
Comment (darrellwarde) [inline .changeset/large-hounds-stare.md:5]: Shouldn't there just be one changeset for this PR?

File: packages/graphql/src/translate/where/property-operations/create-connection-operation.ts [hunk 3]
```diff
@@ -98,3 +99,79 @@ export function createConnectionOperation({
...
+                whereInput: nestedProperties,
+                context,
+                element: edge,
+            });
+
+            return result;
+        }
+
+        if (key.startsWith("node") || key.startsWith(node.name)) {
+            // TODO: improve
```
Comment (darrellwarde) [inline packages/graphql/src/translate/where/property-operations/create-connection-operation.ts:151]: This `TODO` doesn't really offer any value right now - worth adding some context?

**Review by darrellwarde — APPROVED** [review]
LGTM!

File: .changeset/large-hounds-stare.md (hunk 2 above)
Comment (angrykoala) [reply .changeset/large-hounds-stare.md:5]: There are 2 issue fixes here, so I went with 2 changesets, not sure what the appropriate way of doing that is :/

File: packages/graphql/src/translate/where/property-operations/create-connection-operation.ts (hunk 3 above)
Comment (angrykoala) [reply packages/graphql/src/translate/where/property-operations/create-connection-operation.ts:151]:
```suggestion
            // TODO: improve nodeOn properties generation
```

File: .changeset/large-hounds-stare.md (hunk 2 above)
Comment (darrellwarde) [reply .changeset/large-hounds-stare.md:5]: I _think_ the way we use changesets associates one changeset with one PR - so I would document that this PR addresses two issues with a single PR.

Comment (angrykoala) [reply .changeset/large-hounds-stare.md:5]: So, two lines in this changeset and remove the other one then?

Comment (angrykoala) [reply .changeset/large-hounds-stare.md:5]:
```suggestion
Fix nested connection filters using SINGLE and SOME
Fix implicit and parameters missing in connection where
```

**Pull request merged**